spring-integration-file/src/main/java/org/springframework/integration/file/remote/handler/FileTransferringMessageHandler.java

```diff
@@ -139,7 +167,22 @@ protected void onInit() throws Exception {
 
 	@Override
 	protected void handleMessageInternal(Message<?> message) throws Exception {
-		this.remoteFileTemplate.send(message, this.mode);
+		String path = this.remoteFileTemplate.send(message, this.mode);
+		if (this.chmod != null && isChmodCapable()) {
+			doChmod(this.remoteFileTemplate, path, this.chmod);
```

Comment:
But I'm sure that you agree with me that anyone can do the same in his code like my sample and he will really expect "the same session context". But no we retrieve a new instance from the `CCF` with each `remoteFileTemplate` interaction.